src/onewire.rs

```diff
@@ -0,0 +1,135 @@
+//! RMT-based Onewire Implementation
+//!
+//! The Onewire module driver can be used to communicate with onewire (1-Wire)
+//! devices.
+//!
+//! This module is an abstraction around the esp-idf component [onewire_bus](https://components.espressif.com/components/espressif/onewire_bus)
+//! implementation. It is recommended to read the usage of the C API in this [example](https://github.com/espressif/esp-idf/tree/v5.2.2/examples/peripherals/rmt/onewire)
+//!
+//!
+//! This implementation currently supports the one-wire API from the new (v5) esp-idf API.
+//!
+//! The pin this peripheral is attached to must be
+//! externally pulled-up with a 4.7kOhm resistor.
+//!
+//! See the `examples/` folder of this repository for more.
+
+use core::marker::PhantomData;
+use core::ptr;
+
+use esp_idf_sys::*;
+
+use crate::peripheral::Peripheral;
+
+#[derive(Debug)]
+pub struct Device<'b> {
```

Comment:
This will no longer be a driver itself. However, it seems like a device can still implement this through borrowing the `OWDevice` reference to the bus:

```
pub struct DS18B20DeviceDriver<'a, T: Borrow<OWDriver<'a>>>(OWDevice<'a, T>);

impl<'a, T: Borrow<OWDriver<'a>>> DS18B20DeviceDriver<'a, T> {
    pub fn get_temp(&self) -> Result<f32, EspError> {
        self.0.bus().borrow().reset()?;

        self.ds18b20_send_command(Ds18b20Command::ConvertTemp as _);
        let mut buf = [0u8; 10];
        self.0.bus().borrow().read(&mut buf)?;
        println!("buffer = {buf:?}");
        let lsb = buf[0];
        let msb = buf[1];

        let temp_raw: u16 = (u16::from(msb) << 8) | u16::from(lsb);

        Ok(f32::from(temp_raw) / 16.0)
    }

    fn ds18b20_send_command(&self, cmd: u8) -> Result<(), EspError> {
        let mut buf = [0; 10];
        buf[0] = OWCommand::MatchRom as _;
        let addr = self.0.address().to_le_bytes();

        buf[1..9].copy_from_slice(&addr);
        buf[9] = cmd;
        println!(
            "seend command{cmd} to addr {} [{:?}], full_msg = {:?}",
            self.0.address(),
            addr,
            buf
        );
        let write_size = self.0.bus().borrow().write(&buf)?;
        // let write_size = bus.write(&buf)?;

        println!("Sent {write_size} bytes");

        Ok(())
    }
}
```

Reply:
Or just 
```rust
pub struct DS18B20DeviceDriver<'a, T: Borrow<OWDriver<'a>>>(T);
```
?
I.e. no need for the `OWDevice` indirection, as it does not give us anything.